### PR TITLE
Code Improvement for Better Stability

### PR DESCRIPTION
**Pull request by Ankur0812**
2 files changs.

### CODE REVIEW

File: src/Admin/Actions.php [hunk 1]
```diff
@@ -46,7 +46,19 @@ public function register_assets() {
 	}
 
 	public function save_admin_settings() {
-		$post_data = $_POST;
+		//$post_data = $_POST;
```
Comment (mehul0810) [inline src/Admin/Actions.php:49]: @Ankur0812 Remove commenting from all the files

File: assets/src/js/admin/plausible-admin.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
+!(function (e) {
+    var t = {};
+    function n(a) {
```
Comment (mehul0810) [inline assets/src/js/admin/plausible-admin.js:3]: @Ankur0812 Why do we need this file committed with minified version of source?

File: src/Admin/Actions.php [hunk 3]
```diff
@@ -46,7 +46,19 @@ public function register_assets() {
 	}
 
 	public function save_admin_settings() {
-		$post_data = $_POST;
+		//$post_data = $_POST;  
+        $post_data = array();
+		
+		$post_data['action'] = sanitize_text_field(trim($_POST['action']));
+		$post_data['roadblock'] = sanitize_text_field(trim($_POST['roadblock']));
```
Comment (mehul0810) [inline src/Admin/Actions.php:53]: @Ankur0812 Can we create a helper function similar to `wc_clean` to handle this dynamically?

File: src/Admin/Actions.php [hunk 4]
```diff
@@ -55,14 +67,30 @@ public function save_admin_settings() {
...
+		// Save Settings.
+		$response = array(
+			'status'       => $status,
+			'message'      =>  $message,
+		);
 		// Send response.
 		wp_send_json_success(
 			[
-				'message' => esc_html__( 'Settings saved successfully.', 'plausible-analytics' ),
+				'message' => $message,
```
Comment (mehul0810) [inline src/Admin/Actions.php:90]: @Ankur0812 Looks like you missed passing the `$response` array here which you created

File: src/Admin/Settings.php [hunk 5]
```diff
@@ -99,10 +99,13 @@ public function get_header( $name ) {
 	 */
 	public function plausible_analytics_settings_page() {
 		$settings             = Helpers::get_settings();
-		$domain               = ! empty( $settings['domain_name'] ) ? $settings['domain_name'] : Helpers::get_domain();
-		$custom_domain_prefix = ! empty( $settings['custom_domain_prefix'] ) ? $settings['custom_domain_prefix'] : 'analytics';
-		$self_hosted_domain   = ! empty( $settings['self_hosted_domain'] ) ? $settings['self_hosted_domain'] : 'example.com';
-		$shared_link          = ! empty( $settings['shared_link'] ) ? $settings['shared_link'] : "https://plausible.io/share/{$domain}?auth=XXXXXXXXXXXX";
+		//echo "<pre>";
+		//print_r($settings);
```
Comment (mehul0810) [inline src/Admin/Settings.php:103]: @Ankur0812 Remove these debugging comments please

File: src/Admin/Settings.php [hunk 6]
```diff
@@ -99,10 +99,13 @@ public function get_header( $name ) {
...
+		//exit("fdfdd");
+		$domain               = ! empty( $settings['domain_name'] ) ? esc_attr($settings['domain_name']) : Helpers::get_domain();
+		$custom_domain_prefix = ! empty( $settings['custom_domain_prefix'] ) ? esc_attr($settings['custom_domain_prefix']) : 'analytics';
+		$self_hosted_domain   = ! empty( $settings['self_hosted_domain'] ) ? esc_attr($settings['self_hosted_domain']) : 'example.com';
+		$shared_link          = ! empty( $settings['shared_link'] ) ? esc_attr($settings['shared_link']) : "https://plausible.io/share/{$domain}?auth=XXXXXXXXXXXX";
```
Comment (mehul0810) [inline src/Admin/Settings.php:108]: @Ankur0812 This field contains URL. So, use `esc_url` function instead of `esc_attr`

Comment (mehul0810) [reply src/Admin/Settings.php:108]: @Ankur0812 Also, there is no need to add escaping function here. Do late escaping, in which you need to use escape function when you output these variables

File: src/Admin/Settings.php [hunk 7]
```diff
@@ -188,7 +191,7 @@ public function plausible_analytics_settings_page() {
 							<label>
 								<?php esc_html_e( 'Shared Link:', 'plausible-analytics' ); ?>
 								<span class="plausible-analytics-admin-field-input">
-									<input style="width: 550px; max-width: 100%;" type="text" name="plausible_analytics_settings[shared_link]" value="<?php echo $shared_link; ?>" />
+									<input style="width: 550px; max-width: 100%;" type="text" name="plausible_analytics_settings[shared_link]" value="<?php esc_attr_e($shared_link); ?>" />
```
Comment (mehul0810) [inline src/Admin/Settings.php:194]: @Ankur0812 Use `esc_url` instead of `esc_attr_e` as it contains a URL

**Review by mehul0810 — CHANGES_REQUESTED** [review]
@Ankur0812 I have added some comment. Please feel free to implement the changes and let me know if you have any questions.

CC: @metmarkosaric

File: assets/src/js/admin/plausible-admin.js (hunk 2 above)
Comment (Ankur0812) [reply assets/src/js/admin/plausible-admin.js:3]: We need to display Error message on save button.

File: src/Admin/Actions.php (hunk 4 above)
Comment (Ankur0812) [reply src/Admin/Actions.php:84]: Not required, removed

File: src/Admin/Settings.php (hunk 5 above)
Comment (Ankur0812) [reply src/Admin/Settings.php]: Done

File: src/Admin/Settings.php (hunk 6 above)
Comment (Ankur0812) [reply src/Admin/Settings.php]: I already tried but issue is that we escaping later then display prompt window not working.
if you try  then escaping on output variable, then prompt window open.

That's why I escaped above.

File: src/Admin/Settings.php (hunk 7 above)
Comment (Ankur0812) [reply src/Admin/Settings.php]: Done

File: src/Admin/Actions.php (hunk 3 above)
Comment (Ankur0812) [reply src/Admin/Actions.php]: Done

File: src/Admin/Actions.php (hunk 1 above)
Comment (Ankur0812) [reply src/Admin/Actions.php]: Done

**Comment by mehul0810** [comment]
Thanks @Ankur0812 

Merging this in a different branch as it will require some tweaks as well. Also, will share the other PR with @Ankur0812 for learning, if there are essential changes. Thanks!

CC: @metmarkosaric

**Pull request merged**